docs/adapters/adapter_api.md

```diff
@@ -6,9 +6,64 @@ NOTE: This API is currently under active development and is highly likely to cha
 
 __TOC__
 
+## Overview
+
+An Adapter is the interface between the juttle runtime and a backend data source. The backend contains of a set of data points. Each data point contains a number of fields. In most cases, each data point also contains a timestamp.
+
+For reads, the adapter's job is to interpret the options included in the juttle `read` command into a set of matching data points, construct Javascript objects representing those data points, and pass them to the juttle runtime. For writes, the adapter's job is to take the output of programs and "save" that output by writing data points to the backend.
```

Comment:
Something's missing, maybe "to interpret the options included in the juttle `read` command to form a query to the backend, receive the matching data points, construct ..."